js/retry-timeout-callback.js

```diff
@@ -782,6 +818,116 @@ var retryTimeoutCallback = (function() {
         return editorContents;
     };
 
+    var parseParameters = function(editor) {
+        var content = contentManager.getTabbedEditorContents(editor.stepName, bankServiceFileName);
+        // params.retryParms and params.timeoutParms
+        var params = __getParamsFromEditor(content);
+        console.log(params);
+
+        // sort out params and send to playground timeline/browser
+        updatePlayground(editor.stepName, params);
+    };
+
+    var updatePlayground = function(stepName, params) {
+        var browser = contentManager.getBrowser(stepName);
+        browser.setURL(__browserTransactionBaseURL); 
+
+        var retryParams = params.retryParms;
+
+        // Set params, or use default param values
+        // TODO: probably handle setting defaults in __getParamsFromEditor
+        var maxRetries = parseInt(retryParams.maxRetries) || 3;
+        var maxDuration = parseInt(retryParams.maxDuration) || 180000;
+        var delay = parseInt(retryParams.delay) || 0;
+        var jitter = parseInt(retryParams.jitter) || 200;
+        var timeout = parseInt(params.timeoutParms[0]) || 1000;
+
+        var timeoutCount = 0;
+        var elapsedRetryProgress = 0;
+        var $tickContainers = $("[data-step='" + stepName + "']").find('.tickContainer');
+        var timeoutTickContainer = $tickContainers[0];
+        var retryTickContainer = $tickContainers[1];
+        var $progressBar = $("[data-step='" + stepName + "']").find('.progressBar').find('div');
+    
+        // Reset the tick containers for browser refreshes
+        $(timeoutTickContainer).empty();
+        $(retryTickContainer).empty();
+
+        setProgressBar(maxDuration, delay, jitter, timeout, timeoutCount, maxRetries, elapsedRetryProgress, 0, timeoutTickContainer, retryTickContainer, $progressBar);
+    };
+
+    var __getParamsFromEditor = function(content) {
+        var editorContents = {};
+        editorContents.retryParms = {};
+        try {
+            // [0] - original content
+            // [1] - Retry annotation
+            // [2] - retry parameters as a string
+            var retryRegexString = "(@Retry" + "\\s*" + "\\(" + "\\s*" +
+            "((?:\\s*(?:retryOn|maxRetries|maxDuration|durationUnit|delay|delayUnit|jitter|jitterDelayUnit|abortOn)\\s*=\\s*[\\d\.,a-zA-Z]*)*)" +
+            "\\s*" + "\\))";
+
+            // [0] - original content
+            // [1] - Timeout annotation
+            // [2] - parameter value inside parentheses
+            var timeoutRegexString = "\\s*(@Timeout)\\s*" + 
+            "(?:\\(\\s*([\\d]*)\\s*\\))?"; // "(?:(?:unit|value)\\s*=\\s*[\\d\\.,a-zA-Z]+\\s*)*|"
```

Comment:
Is '@Timeout' allowed by itself?  Or do you have to have at least '@Timeout()'?  This regex does not enforce the parenthesis.

It appears that '@Timeout(dd)' does not cause an error....it just allows the default to be 1000 down on line 909 below.  Do we need to check if they have a parm that is not an integer?

Are we allowing for **value** and **unit** to be specified?

Reply:
`@Timeout` is allowed by itself

The integer check is a TODO for now.

We are going to exclude all units for simplicity, but may add support for `value` back in (currently the regex for both are in the comment on the same line)

Reply:
Currently `@Retry` still catches unit parameters but does not take them into account.

TODO: remove units, and throw something like `Unsupported playground parameter or value` in the editor.